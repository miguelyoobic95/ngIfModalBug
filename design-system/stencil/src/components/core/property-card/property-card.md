---
name: Property Card
category: Core Components
---

```property-card-columns.html
    <yoo-property-card></yoo-property-card>
```

```property-card-columns.js
  var comp = document.querySelector('yoo-property-card');
comp.properties = [{
    "title": "Chart title",
    "type": "columns",
    "group": ["group1", "group2"],
    "values": [{
      "title": "label column one",
      "value": "4.1",
      "isPercent": true,
      "color": "balanced"
    }, {
      "title": "label column 2",
      "value": "63,5/76",
      "color": "assertive"
    }, {
      "title": "label column 3",
      "value": "9.3",
      "isPercent": true,
      "color": "assertive"
    }, {
      "title": "label column 4",
      "value": "82.9",
      "isPercent": true,
      "color": "energized"
    }]
  }];
```

```property-card-grid.html
    <yoo-property-card></yoo-property-card>
```
```property-card-grid.js
  var comp = document.querySelector('yoo-property-card');
    comp.properties = [
        {
            "title": "Chart title",
            "type": "grid",
            "group":["group1", "group2"],
            "headers": [
                "Column 1",
                "Column 2",
                "Column 3",
                "Column 4",
                "Column 5"
            ],
            "values": [
                {
                    "values": [
                        "Row 1",
                        "value 1",
                        "value 2",
                        "value 3",
                        null
                    ]
                },
                {
                    "values": [
                        "Row 2",
                        "value 1",
                        "value 2",
                        "value 3",
                        null
                    ]
                },
                {
                    "values": [
                        "Row 3",
                        "value 1",
                        "value 2",
                        "value 3",
                        "Ferm. Except info DC"
                    ],
                    "color": "assertive"
                },
                {
                    "values": [
                        "Row 4",
                        "value 1",
                        "value 2",
                        "value 3",
                        null
                    ]
                },
                {
                    "values": [
                        "Row 5",
                        "value 1",
                        "value 2",
                        "value 3",
                        null
                    ]
                },
                {
                    "values": [
                        "Row 6",
                        "value 1",
                        "value 2",
                        "value 3",
                        null
                    ]
                },
                {
                    "values": [
                        "Row 7",
                        null,
                        null,
                        null,
                        null
                    ]
                }
            ]
        }
    ]
```

```property-card-html.html
    <yoo-property-card></yoo-property-card>
```
```property-card-html.js
  var comp = document.querySelector('yoo-property-card');
comp.properties = [
    {
        "title": "Bloc title",
        "type": "html",
        "group":["group1", "group2"],
        "value": "<center>This is a text</center>"
    }
]
```

```property-card-chart.html
    <yoo-property-card></yoo-property-card>
```
```property-card-chart.js
  var comp = document.querySelector('yoo-property-card');
    comp.properties = [
    {
        "type": "chart",
        "title": "Chart Title",
        "group":["group1", "group2"],
        "config": {
            "chart": {
                "plotBackgroundColor": null,
                "plotBorderWidth": 0,
                "plotShadow": false
            },
            "title": {
                "text": "Browser<br>shares<br>2015",
                "align": "center",
                "verticalAlign": "middle",
                "y": 40
            },
            "tooltip": {
                "pointFormat": "{series.name}: <b>{point.percentage:.1f}%</b>"
            },
            "plotOptions": {
                "pie": {
                    "dataLabels": {
                        "enabled": true,
                        "distance": -50,
                        "style": {
                            "fontWeight": "bold",
                            "color": "red"
                        }
                    },
                    "startAngle": -90,
                    "endAngle": 90,
                    "center": [
                        "50%",
                        "75%"
                    ]
                }
            },
            "series": [
                {
                    "type": "pie",
                    "name": "Share",
                    "innerSize": "50%",
                    "data": [
                        [
                            "Value 1",
                            10.38
                        ],
                        [
                            "Value 2",
                            56.33
                        ],
                        [
                            "Value 3",
                            24.03
                        ],
                        [
                            "Value 4",
                            4.77
                        ],
                        [
                            "Value 5",
                            0.91
                        ],
                        {
                            "name": "Proprietary or Undetectable",
                            "y": 0.2,
                            "dataLabels": {
                                "enabled": false
                            }
                        }
                    ]
                }
            ]
        }
    }
]
```

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`properties`|any|*required| the properties of the card|
