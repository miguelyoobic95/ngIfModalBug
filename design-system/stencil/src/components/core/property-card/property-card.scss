:host {
  .details {
    border-top: $border-width solid cssvar(border-color);
    border-bottom: $border-width solid cssvar(border-color);
    background: cssvar(light);
    font-size: cssvar(font-medium);
    display: table;
    width: 100%;
    font-size: cssvar(font-medium);
    margin-bottom: $padding/2;

    .details-title {
      font-weight: 500;
      padding: $padding/2;
      margin-bottom: $padding/4;
      border-bottom: $border-width solid cssvar(border-color);
      text-align: center;
      text-transform: uppercase;
    }
    .description {
      color: cssvar(text-header);
      padding: $padding/2; // 0;
      line-height: cssvar(font-xlarge);
      &.tags {
        margin-bottom: -5px;
        >span {
          @include tags;
          margin-bottom: 5px;
          padding: 3px 8px;
        }
      }
    }
  }
  .p-type-columns {
    .p-column {
      &:not(:last-child) {
        border-right: $border-width solid cssvar(border-color);
      }
      .title {
        font-size: cssvar(font-small);
        font-weight: 400;
        text-align: center;
        line-height: 11px;
        padding: 0 $padding/2;
      }
      .number {
        font-weight: 500;
        font-size: cssvar(font-large);
        margin-top: $padding/4;
      }
      .circle {
        width: 10px;
        height: 10px;
        border-radius: 50%;
        margin-top: 5px;
      }
    }
  }
  .p-type-rows,
  .p-type-grid {
    .p-row {
      padding: $padding/4 0;
      &:first-child {
        padding-top: 0;
      }
      &:last-child {
        padding-bottom: 0;
      }
      &:not(:last-child) {
        border-bottom: $border-width solid cssvar(border-color);
      }
      .title {
        font-size: cssvar(font-small);
        font-weight: 400;
        width: 150px;
        @include text-truncate();
      }
      .number {
        font-weight: 400; //font-size: cssvar(font-large);
      }
      .circle {
        width: 10px;
        height: 10px;
        border-radius: 50%;
        text-align: center;
        margin-top: $padding/4;
      }
      .delta {
        text-align: right;
        padding-right: $padding/2;
      }
    }
  }
  .p-type-grid {
    .p-row {
      >div {
        @include text-truncate();
        font-size: cssvar(font-small);
        font-weight: 400;
      }
      >div:not(:last-child) {
        padding-right: $padding/4;
      }
      &.header>div {
        @include text-truncate();
        font-weight: 500;
        text-transform: uppercase;
        font-size: cssvar(font-xsmall);
      }
      .break-lines {
        white-space: normal;
      }
    }
  }
  .p-type-chart {
    chart-high {
      height: 300px;
    }
  }
}
