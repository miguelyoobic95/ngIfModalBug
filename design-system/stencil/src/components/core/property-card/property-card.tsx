import { Component, Element, Prop, State } from '@stencil/core';
import { ITranslateService } from '@shared/interfaces';

@Component({
    tag: 'yoo-property-card',
    styleUrl: 'property-card.scss',
    scoped: true
})
export class YooPropertyCardComponent {

    @Prop() properties: any;

    @Element() host: HTMLStencilElement;

    @State() isInit: boolean;

    private translate: ITranslateService = (window as any).translateService;

    componentDidLoad() {
        // let defaultConfig = {
        //     chart: {
        //         style: { fontFamily: 'Visby' },
        //         backgroundColor: null,
        //         height: '300px'
        //     },
        //     credits: { enabled: false },
        //     exporting: { enabled: false },
        //     tooltip: { followTouchMove: false }
        // };
        // if (this.properties) {
        //     this.properties.forEach(p => {
        //         if (p.type === 'chart') {
        //             // p.config = merge(cloneDeep(p.config), defaultConfig);
        //         }
        //     });
        // }
        this.translate = (window as any).translateService;

        this.isInit = true;
    }

    isString(value) {
        return typeof value === 'string';
    }

    onChartMoving($event) {
    }

    isVisible(p: any) {
        if (!p.group && !p.role) {
            return true;
        }
        let authorized = true;
        if (p.group) {
            // authorized = this.authentication.hasGroups([].concat(p.group));
        }
        if (p.role) {
            // authorized = this.authentication.hasRoles([].concat(p.role));
        }
        return authorized;
    }

    renderColumns(p: any): JSX.Element {
        return <div>
            {p.values.map((value) =>
                <div class="p-column">
                    <div class="number">
                        {(value.value ? <span>{value.value}</span> : null)}
                        {(value.isPercent ? <span>%</span> : null)}
                    </div>
                    {(value.title ?
                        <div class="title">
                            <span>{(this.translate ? this.translate.get(value.title) : value.title)}</span>
                        </div>
                        : null)}
                    <div class={'circle bg-' + value.color}></div>
                </div>
            )}
        </div>;
    }

    renderGrid(p: any): JSX.Element {
        return <div>
            <div class="p-row header">
                {p.headers.map((value) =>
                    (value.title ? <span class={(value.truncate ? 'break-lines' : null)}>{(this.translate ? this.translate.get(value.title) : value.title)}</span> : null)
                )}
            </div>
            {p.values.map((r) =>
                <div class={'p-row ' + r.color}>
                    {r.values.map((value) =>
                        (value ? <span class={(value.truncate ? 'break-lines' : null)}> {(this.isString(value) ? value : (this.translate ? this.translate.get(value.title) : value.title)) } </span> : null)
                    )}
                </div>
            )}
        </div>;
    }

    renderRows(p: any): JSX.Element {
        return <div>
            {p.values.map((value) =>
                <div class="p-row">
                    {(value.title ?
                        <div class="title">
                            <span>{(this.translate ? this.translate.get(value.title) : value.title)}</span>
                        </div>
                        : null)}
                    {(value.value ?
                        <div class="number">
                            <span>{value.value}</span>
                        </div>
                        : null)}
                    {(value.delta ?
                        <div class="delta">
                            <span>{value.delta}</span>
                        </div>
                        : null)}
                    {(value.color ?
                        <div class={'circle bg-' + value.color}></div>
                        : null)}
                </div>
            )}
        </div>;
    }

    renderChart(p: any): JSX.Element {
        return <div></div>;
        // return <chart-high config={p.config} hide-title={true} (moving)="onChartMoving($event)"></chart-high>;
    }

    renderHtml(p: any): JSX.Element {
        return <div>{p.value}</div>;
    }

    renderProperty(p: any): JSX.Element {
        return (this.isVisible(p) ?
            <div class="details">
                {(p.title ? <div class="details-title">{(this.translate ? this.translate.get(p.title) : p.title)}</div> : null)}
                {(p.type ?
                    <div class={'description p-type-' + p.type}>
                        {(p.type === 'columns' ? this.renderColumns(p) : null)}
                        {(p.type === 'grid' ? this.renderGrid(p) : null)}
                        {(p.type === 'rows' ? this.renderRows(p) : null)}
                        {(p.type === 'chart' ? this.renderChart(p) : null)}
                        {(p.type === 'html' ? this.renderHtml(p) : null)}
                    </div>
                    : null)}
            </div>
            : null);
    }

    render(): JSX.Element {
        return ((this.isInit && this.properties) ?
            <div>
                {this.properties.map((property) =>
                    this.renderProperty(property)
                )}
            </div>
            : null);
    }
}
