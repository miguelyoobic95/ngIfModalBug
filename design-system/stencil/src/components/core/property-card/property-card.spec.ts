import { TestWindow } from '@stencil/core/testing';
import { YooPropertyCardComponent } from './property-card';

describe('PropertyCardComponent', () => {
    const propertiesColumns = [
        {
            'title': 'Chart title',
            'type': 'columns',
            'group': ['group1', 'group2'],
            'values': [
                {
                    'title': 'label column one',
                    'value': '4.1',
                    'isPercent': true,
                    'color': 'balanced'
                },
                {
                    'title': 'label column 2',
                    'value': '63,5/76',
                    'color': 'assertive'
                },
                {
                    'title': 'label column 3',
                    'value': '9.3',
                    'isPercent': true,
                    'color': 'assertive'
                },
                {
                    'title': 'label column 4',
                    'value': '82.9',
                    'isPercent': true,
                    'color': 'energized'
                }
            ]
        }
    ];
    const propertiesGrid = [
        {
            'title': 'Chart title',
            'type': 'grid',
            'group': ['group1', 'group2'],
            'headers': [
                'Column 1',
                'Column 2',
                'Column 3',
                'Column 4',
                'Column 5'
            ],
            'values': [
                {
                    'values': [
                        'Row 1',
                        'value 1',
                        'value 2',
                        'value 3',
                        null
                    ]
                },
                {
                    'values': [
                        'Row 2',
                        'value 1',
                        'value 2',
                        'value 3',
                        null
                    ]
                },
                {
                    'values': [
                        'Row 3',
                        'value 1',
                        'value 2',
                        'value 3',
                        'Ferm. Except info DC'
                    ],
                    'color': 'assertive'
                },
                {
                    'values': [
                        'Row 4',
                        'value 1',
                        'value 2',
                        'value 3',
                        null
                    ]
                },
                {
                    'values': [
                        'Row 5',
                        'value 1',
                        'value 2',
                        'value 3',
                        null
                    ]
                },
                {
                    'values': [
                        'Row 6',
                        'value 1',
                        'value 2',
                        'value 3',
                        null
                    ]
                },
                {
                    'values': [
                        'Row 7',
                        null,
                        null,
                        null,
                        null
                    ]
                }
            ]
        }
    ];
    const propertiesChart = [
        {
            'type': 'chart',
            'title': 'Chart Title',
            'group': ['group1', 'group2'],
            'config': {
                'chart': {
                    'plotBackgroundColor': null,
                    'plotBorderWidth': 0,
                    'plotShadow': false
                },
                'title': {
                    'text': 'Browser<br>shares<br>2015',
                    'align': 'center',
                    'verticalAlign': 'middle',
                    'y': 40
                },
                'tooltip': {
                    'pointFormat': '{series.name}: <b>{point.percentage:.1f}%</b>'
                },
                'plotOptions': {
                    'pie': {
                        'dataLabels': {
                            'enabled': true,
                            'distance': -50,
                            'style': {
                                'fontWeight': 'bold',
                                'color': 'red'
                            }
                        },
                        'startAngle': -90,
                        'endAngle': 90,
                        'center': [
                            '50%',
                            '75%'
                        ]
                    }
                },
                'series': [
                    {
                        'type': 'pie',
                        'name': 'Share',
                        'innerSize': '50%',
                        'data': [
                            [
                                'Value 1',
                                10.38
                            ],
                            [
                                'Value 2',
                                56.33
                            ],
                            [
                                'Value 3',
                                24.03
                            ],
                            [
                                'Value 4',
                                4.77
                            ],
                            [
                                'Value 5',
                                0.91
                            ],
                            {
                                'name': 'Proprietary or Undetectable',
                                'y': 0.2,
                                'dataLabels': {
                                    'enabled': false
                                }
                            }
                        ]
                    }
                ]
            }
        }
    ];
    const propertiesHtml = [
        {
            'title': 'Bloc title',
            'type': 'html',
            'group': ['group1', 'group2'],
            'value': '<center>This is a text</center>'
        }
    ];

    it('should build', () => {
        expect(new YooPropertyCardComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        let element, window;
        beforeEach(async () => {
            window = new TestWindow();
            element = await window.load({
                components: [YooPropertyCardComponent],
                html: '<yoo-property-card></yoo-property-card>'
            });
        });

        it('Should render', async () => {
            expect(element).toMatchSnapshot();
        });
        it('Should render as columns', async () => {
            element.properties = propertiesColumns;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render as grid', async () => {
            element.properties = propertiesGrid;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render as chart', async () => {
            element.properties = propertiesChart;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render as html', async () => {
            element.properties = propertiesHtml;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
    });
});