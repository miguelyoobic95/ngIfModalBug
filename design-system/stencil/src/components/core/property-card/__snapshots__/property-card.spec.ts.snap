// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCardComponent Rendering Should render 1`] = `
<yoo-property-card
  class="hydrated"
  data-yoo-property-card-host=""
>
  
  
</yoo-property-card>
`;

exports[`PropertyCardComponent Rendering Should render as chart 1`] = `
<yoo-property-card
  class="hydrated"
  data-yoo-property-card-host=""
>
  
  <div
    data-yoo-property-card=""
  >
    <div
      class="details"
      data-yoo-property-card=""
    >
      <div
        class="details-title"
        data-yoo-property-card=""
      >
        Chart Title
      </div>
      <div
        class="description p-type-chart"
        data-yoo-property-card=""
      >
        
        <div
          data-yoo-property-card=""
        />
        
      </div>
    </div>
  </div>
</yoo-property-card>
`;

exports[`PropertyCardComponent Rendering Should render as columns 1`] = `
<yoo-property-card
  class="hydrated"
  data-yoo-property-card-host=""
>
  
  <div
    data-yoo-property-card=""
  >
    <div
      class="details"
      data-yoo-property-card=""
    >
      <div
        class="details-title"
        data-yoo-property-card=""
      >
        Chart title
      </div>
      <div
        class="description p-type-columns"
        data-yoo-property-card=""
      >
        <div
          data-yoo-property-card=""
        >
          <div
            class="p-column"
            data-yoo-property-card=""
          >
            <div
              class="number"
              data-yoo-property-card=""
            >
              <span
                data-yoo-property-card=""
              >
                4.1
              </span>
              <span
                data-yoo-property-card=""
              >
                %
              </span>
            </div>
            <div
              class="title"
              data-yoo-property-card=""
            >
              <span
                data-yoo-property-card=""
              >
                label column one
              </span>
            </div>
            <div
              class="circle bg-balanced"
              data-yoo-property-card=""
            />
          </div>
          <div
            class="p-column"
            data-yoo-property-card=""
          >
            <div
              class="number"
              data-yoo-property-card=""
            >
              <span
                data-yoo-property-card=""
              >
                63,5/76
              </span>
              
            </div>
            <div
              class="title"
              data-yoo-property-card=""
            >
              <span
                data-yoo-property-card=""
              >
                label column 2
              </span>
            </div>
            <div
              class="circle bg-assertive"
              data-yoo-property-card=""
            />
          </div>
          <div
            class="p-column"
            data-yoo-property-card=""
          >
            <div
              class="number"
              data-yoo-property-card=""
            >
              <span
                data-yoo-property-card=""
              >
                9.3
              </span>
              <span
                data-yoo-property-card=""
              >
                %
              </span>
            </div>
            <div
              class="title"
              data-yoo-property-card=""
            >
              <span
                data-yoo-property-card=""
              >
                label column 3
              </span>
            </div>
            <div
              class="circle bg-assertive"
              data-yoo-property-card=""
            />
          </div>
          <div
            class="p-column"
            data-yoo-property-card=""
          >
            <div
              class="number"
              data-yoo-property-card=""
            >
              <span
                data-yoo-property-card=""
              >
                82.9
              </span>
              <span
                data-yoo-property-card=""
              >
                %
              </span>
            </div>
            <div
              class="title"
              data-yoo-property-card=""
            >
              <span
                data-yoo-property-card=""
              >
                label column 4
              </span>
            </div>
            <div
              class="circle bg-energized"
              data-yoo-property-card=""
            />
          </div>
        </div>
        
      </div>
    </div>
  </div>
</yoo-property-card>
`;

exports[`PropertyCardComponent Rendering Should render as grid 1`] = `
<yoo-property-card
  class="hydrated"
  data-yoo-property-card-host=""
>
  
  <div
    data-yoo-property-card=""
  >
    <div
      class="details"
      data-yoo-property-card=""
    >
      <div
        class="details-title"
        data-yoo-property-card=""
      >
        Chart title
      </div>
      <div
        class="description p-type-grid"
        data-yoo-property-card=""
      >
        
        <div
          data-yoo-property-card=""
        >
          <div
            class="p-row header"
            data-yoo-property-card=""
          >
            
          </div>
          <div
            class="p-row undefined"
            data-yoo-property-card=""
          >
            <span
              class=""
              data-yoo-property-card=""
            >
               Row 1 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 1 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 2 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 3 
            </span>
            
          </div>
          <div
            class="p-row undefined"
            data-yoo-property-card=""
          >
            <span
              class=""
              data-yoo-property-card=""
            >
               Row 2 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 1 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 2 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 3 
            </span>
            
          </div>
          <div
            class="p-row assertive"
            data-yoo-property-card=""
          >
            <span
              class=""
              data-yoo-property-card=""
            >
               Row 3 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 1 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 2 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 3 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               Ferm. Except info DC 
            </span>
          </div>
          <div
            class="p-row undefined"
            data-yoo-property-card=""
          >
            <span
              class=""
              data-yoo-property-card=""
            >
               Row 4 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 1 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 2 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 3 
            </span>
            
          </div>
          <div
            class="p-row undefined"
            data-yoo-property-card=""
          >
            <span
              class=""
              data-yoo-property-card=""
            >
               Row 5 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 1 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 2 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 3 
            </span>
            
          </div>
          <div
            class="p-row undefined"
            data-yoo-property-card=""
          >
            <span
              class=""
              data-yoo-property-card=""
            >
               Row 6 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 1 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 2 
            </span>
            <span
              class=""
              data-yoo-property-card=""
            >
               value 3 
            </span>
            
          </div>
          <div
            class="p-row undefined"
            data-yoo-property-card=""
          >
            <span
              class=""
              data-yoo-property-card=""
            >
               Row 7 
            </span>
            
          </div>
        </div>
        
      </div>
    </div>
  </div>
</yoo-property-card>
`;

exports[`PropertyCardComponent Rendering Should render as html 1`] = `
<yoo-property-card
  class="hydrated"
  data-yoo-property-card-host=""
>
  
  <div
    data-yoo-property-card=""
  >
    <div
      class="details"
      data-yoo-property-card=""
    >
      <div
        class="details-title"
        data-yoo-property-card=""
      >
        Bloc title
      </div>
      <div
        class="description p-type-html"
        data-yoo-property-card=""
      >
        
        <div
          data-yoo-property-card=""
        >
          &lt;center&gt;This is a text&lt;/center&gt;
        </div>
      </div>
    </div>
  </div>
</yoo-property-card>
`;
